### PR TITLE
enh(Subnav): Add remaining attributed logos to TitleLink logo dict

### PR DESCRIPTION
**Pull request by EnMod**
🎟️ [Asana Task](https://app.asana.com/0/346384260719996/1201461699398493/f)
🔍 [Preview Link](https://react-components-git-nqsubnav-add-attributed-logos-hashicorp.vercel.app/components/subnav)

---

## Description

This PR adds attributed logo options to the logo dictionary in `<TitleLink />`.

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [x] Add Asana and Preview links above.
- [x] Conduct thorough self-review.
- [x] Add or update tests as appropriate.
- [x] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [x] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [x] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

**Review by alexcarpenter — COMMENTED** [review]
Something to consider is that the attributed logos need a bit more height to make the HashiCorp text legible as seen here https://github.com/hashicorp/react-components/blob/727320494c1e618071f9c7acee68f2b066d7246c/packages/subnav/partials/TitleLink/style.module.css#L25-L30

**Comment by EnMod** [comment]
Ah yep, good catch @alexcarpenter :+1: Just updated to include the new logos in that CSS rule

File: packages/subnav/partials/TitleLink/style.module.css [hunk 1]
```diff
@@ -23,7 +23,13 @@
...
 
   &.hashiCorpVault,
-  &.hashiCorpConsul {
+  &.hashiCorpConsul,
+  &.hashiCorpNomad,
+  &.hashiCorpPacker,
+  &.hashiCorpTerraform,
+  &.hashiCorpBoundary,
+  &.hashiCorpWaypoint,
+  &.hashiCorpTfc {
```
Comment (alexcarpenter) [inline packages/subnav/partials/TitleLink/style.module.css:32]: @EnMod  missing `hashiCorpVagrant`

Comment (EnMod) [reply packages/subnav/partials/TitleLink/style.module.css:33]: 😅 done

**Review by alexcarpenter — APPROVED** [review]
👍🏼

**Pull request merged**